pandas/core/indexing.py

```diff
@@ -830,6 +835,11 @@ def _ensure_listlike_indexer(self, key, axis=None, value=None) -> None:
 
     @final
     def __setitem__(self, key, value) -> None:
+        if using_copy_on_write():
+            # print("_LocationIndexer.__setitem__ refcount: ",sys.getrefcount(self.obj))
+            if sys.getrefcount(self.obj) <= 2:
```

Comment:
Why is this 2 instead of 3?

Reply:
I am not fully sure. The code path is a bit different: the Series/DataFrame object is at this point only stored in `self.obj`, while in case of `Series/DataFrame.__setitem__`, it is also only stored as `self`, but `self` also passed as argument to the method being called (`self` is in the signature). My assumption is that this aspect influences the reference count, but I am not 100% sure about that (that is also the reason that it is 2 and not 1 here, since by passing it to `sys.getrefcount`, inside that function another reference is always made).

Reply:
Thx